rocks-db/src/clients/asset_client.rs

```diff
@@ -314,19 +324,129 @@ impl Storage {
         })
     }
 
+    pub async fn get_master_edition_child_assets(
+        &self,
+        master_edition: Pubkey,
+        pagination: PaginationQuery,
+    ) -> Result<Vec<TokenMetadataEditionParentIndex>> {
+        let limit = pagination.limit.unwrap_or(100) as usize;
+        let mut revert = false;
+
+        let iter = if pagination.after.is_some() {
+            let mut iter = self.token_metadata_edition_parent_index.iter(EditionIndexKey {
+                pub_key: master_edition,
+                edition: pagination
+                    .after
+                    .unwrap()
+                    .parse::<u64>()
+                    .expect("failed to parse edition after key"),
+            });
+            // iterator is on the item we were searching for
+            iter.next();
+            iter
+        } else if pagination.before.is_some() {
+            let mut iter = self.token_metadata_edition_parent_index.iter_reverse(EditionIndexKey {
+                pub_key: master_edition,
+                edition: pagination
+                    .before
+                    .unwrap()
+                    .parse::<u64>()
+                    .expect("failed to parse edition before key"),
+            });
+            revert = true;
+
+            // iterator is on the item we were searching for
+            iter.next();
+            iter
+        } else {
+            let page = pagination.page.unwrap_or(1) - 1;
+            let mut iter = self.db.prefix_iterator_cf(
+                &self.db.cf_handle(TokenMetadataEditionParentIndex::NAME).unwrap(),
+                master_edition.as_ref(),
+            );
+
+            if page > 0 {
+                let skip = page as usize * limit;
+                // iter.skip(n) changes the iterator type, making it incompatible with RocksDB.
+                for _ in 0..skip {
+                    iter.next();
+                }
+            }
+
+            iter
+        };
+
+        let mut asset_keys: Vec<TokenMetadataEditionParentIndex> = Vec::with_capacity(limit);
```

Comment:
If you already know the size, maybe is it better not to use heap-allocated structures? Seems like an array could be used.

Reply:
maximum limit is 100
I'm not sure that this will bring any growth but at the same time the code itself will become more complicated. 
because we will have Vec<Option<TokenMetadataEditionParentIndex>> and some other things that required additional checks